bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -272,14 +273,31 @@ fn main() -> Result<()> {
                                 TelemetryRecord::PlatformRuntimeDone {
                                     request_id, status, ..
                                 } => {
+                                    if status != Status::Success {
+                                        match lambda_enhanced_metrics.increment_errors_metric() {
+                                            Ok(()) => {}
+                                            Err(e) => {
+                                                error!("Failed to increment error metric: {e:?}");
+                                            }
+                                        }
+                                        if status == Status::Timeout {
+                                            match lambda_enhanced_metrics.increment_timeout_metric()
+                                            {
+                                                Ok(()) => {}
+                                                Err(e) => {
+                                                    error!(
+                                                        "Failed to increment timeout metric: {e:?}"
+                                                    );
+                                                }
+                                            }
+                                        }
+                                    }
```

Comment:
Too many `match`es for code where we aren't doing anything with `Ok`, can't we just do
```rust
if let Err(e) = lambda_enhanced_metrics.increment_errors_metric() {
        error!("Failed to increment errors metric: {e:?}");
};
```